x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -36,25 +36,32 @@ type scheduler struct {
 	state     *state
 	log       *logp.Logger
 	limiter   *limiter
+	metrics   *inputMetrics
 }
 
 // newScheduler, returns a new scheduler instance
 func newScheduler(publisher cursor.Publisher, bucket *storage.BucketHandle, src *Source, cfg *config,
 	state *state, log *logp.Logger,
 ) *scheduler {
-	return &scheduler{
+	s := scheduler{
 		publisher: publisher,
 		bucket:    bucket,
 		src:       src,
 		cfg:       cfg,
 		state:     state,
 		log:       log,
 		limiter:   &limiter{limit: make(chan struct{}, src.MaxWorkers)},
+		metrics:   newInputMetrics(src.BucketName, nil),
 	}
+	s.metrics.url.Set("gs://" + s.src.BucketName)
```

Comment:
I think this can be moved into `newInputMetrics` given that you are passing the bucket name to it.